### PR TITLE
feat: notify service

### PR DESCRIPTION
**Pull request by quake**
This PR resolve https://github.com/nervosnetwork/ckb/issues/1860 and refactor network alert script notification by adding a notify service, and it's required to implement https://github.com/nervosnetwork/ckb/issues/1867

**configuration  file breaking change**

```diff
-# [alert_notifier]
-# # Script will be notified when node received an alert, first arg is alert message string.
-# notify_script = "echo"
+# [notifier]
+# # Execute command when the new tip block changes, first arg is block struct in json format string.
+# new_block_notify_script = "your_new_block_notify_script.sh"
+# # Execute command when node received an network alert, first arg is alert message string.
+# network_alert_notify_script = "your_network_alert_notify_script.sh"
```

### CODE REVIEW

File: notify/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+                // notify script
+                if let Some(script) = self.config.new_block_notify_script.as_ref() {
+                    let block_json: ckb_jsonrpc_types::BlockView = block.into();
+                    let args =
+                        [serde_json::to_string(&block_json).expect("serializing should be ok")];
+                    if let Err(err) = Command::new(script).args(&args).status() {
+                        error!(
+                            "failed to run new_block_notify_script script, error: {}",
+                            err
+                        );
```
Comment (keroro520) [inline notify/src/lib.rs]: It's better to log the `command` and `args` also.

File: chain/src/chain.rs [hunk 2]
```diff
@@ -385,6 +385,10 @@ impl ChainService {
                     error!("notify new_uncle error {}", e);
                 }
             }
+            let block_ref: &BlockView = &block;
+            self.shared
+                .notify_controller()
+                .notify_new_block(block_ref.clone());
```
Comment (jjyr) [inline chain/src/chain.rs]:
```suggestion
                .notify_new_block(block);
```

File: notify/src/lib.rs (hunk 1 above)
Comment (quake) [reply notify/src/lib.rs]: Added `command` to the error log, `args` is too large, was skipped. Please review again, thanks.

File: notify/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            Ok(block) => {
+                trace!("event new block {:?}", block);
+                // notify all subscribers
+                for subscriber in self.new_block_subscribers.values() {
+                    let _ = subscriber.send(block.clone());
+                }
+                // notify script
+                if let Some(script) = self.config.new_block_notify_script.as_ref() {
+                    let block_json: ckb_jsonrpc_types::BlockView = block.into();
+                    let args = [to_string(&block_json).expect("serializing should be ok")];
```
Comment (jjyr) [inline notify/src/lib.rs]: A block may be very large like a few hundred KB, it may exceed the limits of Linux. we should pass the block hash instead.

Comment (quake) [reply notify/src/lib.rs]: Updated, please review again, thanks

File: chain/src/chain.rs (hunk 2 above)
Comment (quake) [reply chain/src/chain.rs]: I prefer breaking change in the configuration file than maintaining multiple conf version compatibility.

**Pull request merged**